digits/model/tasks/caffe_train.py

```diff
@@ -1256,15 +1257,16 @@ def infer_many_images(self, images, snapshot_epoch=None, gpu=None):
             for index, image in enumerate(chunk):
                 net.blobs['data'].data[index] = self.get_transformer().preprocess(
                         'data', image)
-            o = net.forward()
-            if outputs is None:
-                outputs = o
-            else:
-                for name,blob in o.iteritems():
-                    outputs[name] = np.vstack((outputs[name], blob))
-            print 'Processed %s/%s images' % (len(outputs[outputs.keys()[0]]), len(caffe_images))
+            outputs = net.forward()
+            for output_name in net.outputs:
+                blob = outputs[output_name]
+                if output_name in results:
+                    results[output_name] = np.vstack((results[output_name], blob))
```

Comment:
This is working for me, thanks!

Question: I was wondering why you didn't iterate with `for name,blob in net.forward().iteritems()`, which felt more concise and easy to read. I understand that `net.forward()` returns an unordered dictionary while `net.outputs` is an (ordered) array and this is why you need to iterate over `net.outputs`, right?